include/seqan3/range/views/minimiser.hpp

```diff
@@ -407,26 +506,60 @@ struct minimiser_fn
-        static_assert(std::ranges::forward_range<urng_t>,
-            "The range parameter to views::minimiser must model std::ranges::forward_range.");
+        static_assert(std::ranges::viewable_range<urng1_t>,
+                      "The range parameter to views::minimiser cannot be a temporary of a non-view range.");
+        static_assert(std::ranges::forward_range<urng1_t>,
+                      "The range parameter to views::minimiser must model std::ranges::forward_range.");
 
-        if (window_values_size == 1)
-            throw std::invalid_argument{"The chosen window_valzes_size is not valid. "
-                                        "Please choose a value greater than 1."};
+        if (window_values_size == 1) // Would just return urange1 without any changes
+            throw std::invalid_argument{"The chosen window_values_size is not valid. "
+                                        "Please choose a value greater than 1 or use two ranges."};
 
-        return minimiser_view{urange, window_values_size};
+        return minimiser_view{urange1, window_values_size};
+    }
+
+    //!\brief Store the number of values in one window and the second range and return a range adaptor closure object.
+    template <std::ranges::range urng2_t>
+    constexpr auto operator()(size_t const window_values_size, urng2_t && urange2) const
+    {
+        return adaptor_from_functor{*this, window_values_size, urange2};
+    }
+
+    /*!\brief Call the view's constructor with two arguments: the underlying view and an integer indicating how many
+     *        values one window contains.
+     * \param[in] urange1 The input range to process. Must model std::ranges::viewable_range and
+     *                    std::ranges::forward_range.
+     * \param[in] window_values_size The number of values in one window.
+     * \param[in] urange2 The second input range to process. Must model std::ranges::viewable_range and
+     *                    std::ranges::forward_range.
+     * \returns A range of converted values.
+     */
+    template <std::ranges::range urng1_t, std::ranges::range urng2_t>
+    constexpr auto operator()(urng1_t && urange1, size_t const window_values_size, urng2_t && urange2) const
+    {
+        static_assert(std::ranges::viewable_range<urng1_t>,
+                      "The range1 parameter to views::minimiser cannot be a temporary of a non-view range.");
+        static_assert(std::ranges::viewable_range<urng2_t>,
+                      "The range2 parameter to views::minimiser cannot be a temporary of a non-view range.");
+        static_assert(std::ranges::forward_range<urng1_t>,
+                      "The range1 parameter to views::minimiser must model std::ranges::forward_range.");
+        static_assert(std::ranges::forward_range<urng2_t>,
+                      "The range2 parameter to views::minimiser must model std::ranges::forward_range.");
+
+        if (std::ranges::size(urange1) != std::ranges::size(urange2))
```

Comment:
This line is my guess why #1754 not finished yet.

I think the problem is that you can't assume sized-ness from any range and this expression breakes for filter views, because they don't know their size.